### PR TITLE
Update api document

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: work/v1/types.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ type ManifestCondition struct {
 type ManifestConditionType string
 
 const (
-	// ManifestProgressing represents the resource is being applied on the managed cluster
+	// ManifestProgressing represents the resource the is being applied on the managed cluster
```
Comment (cadawson) [inline work/v1/types.go]: This was my error. This should be "that", not "the".

File: cluster/v1/types.go [hunk 2]
```diff
@@ -22,11 +22,11 @@ import (
...
 // The cluster join process follows a double opt-in process:
 //
-// 1. agent on managed cluster creates CSR on hub with cluster UID and agent name.
-// 2. agent on managed cluster creates ManagedCluster on hub.
-// 3. cluster admin on hub approves the CSR for the ManagedCluster's UID and agent name.
-// 4. cluster admin sets spec.acceptClient of ManagedCluster to true.
-// 5. cluster admin on managed cluster creates credential of kubeconfig to hub.
+// 1. Agent on managed cluster creates CSR on hub with cluster UID and agent name.
+// 2. Agent on managed cluster creates ManagedCluster on hub.
+// 3. Cluster admin on hub approves the CSR for the ManagedCluster's UID and agent name.
```
Comment (cadawson) [inline cluster/v1/types.go]: Would be a good idea to remove the possessive here. Let's change:
"Cluster admin on hub approves the CSR for the ManagedCluster's UID and agent name." to 
"Cluster admin on hub approves the CSR for the UID and agent name of the ManagedCluster."

File: cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml [hunk 3]
```diff
@@ -34,10 +34,10 @@ spec:
       description: "ManagedCluster represents the desired state and current status
         of managed cluster. ManagedCluster is a cluster scoped resource. The name
         is the cluster UID. \n The cluster join process follows a double opt-in process:
-        \n 1. agent on managed cluster creates CSR on hub with cluster UID and agent
-        name. 2. agent on managed cluster creates ManagedCluster on hub. 3. cluster
+        \n 1. Agent on managed cluster creates CSR on hub with cluster UID and agent
+        name. 2. Agent on managed cluster creates ManagedCluster on hub. 3. Cluster
```
Comment (cadawson) [inline cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml:38]: Remove possessive here by replacing:
"Cluster admin on hub approves the CSR for the ManagedCluster's UID and agent name."
with 
"Cluster admin on the hub approves the CSR for the UID and agent name of the ManagedCluster."

File: cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml [hunk 4]
```diff
@@ -34,10 +34,10 @@ spec:
...
         admin on hub approves the CSR for the ManagedCluster's UID and agent name.
-        4. cluster admin sets spec.acceptClient of ManagedCluster to true. 5. cluster
+        4. Cluster admin sets spec.acceptClient of ManagedCluster to true. 5. Cluster
         admin on managed cluster creates credential of kubeconfig to hub. \n Once
         the hub creates the cluster namespace, the Klusterlet agent on the Managed
         Cluster pushes the credential to the hub to use against the managed cluster's
```
Comment (cadawson) [inline cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml]: Remove possessive. Suggestion to replace:
"Cluster pushes the credential to the hub to use against the managed cluster's kube-apiserver."
with
"Cluster pushes the credential to the hub to use with the kube-apiserver of the managed cluster."

**Review by cadawson — APPROVED** [review]
Thank you, Jian Qiu. I made a couple of suggestions, but I kept it to a few. I will go ahead and approve here, in advance. Our squad is going to take a look at these more often to make sure that they are more consistent across the product. I can make the PR a little more friendly next time.

File: cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml (hunk 3 above)
Comment (qiujian16) [reply cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml:38]: done

File: work/v1/types.go (hunk 1 above)
Comment (qiujian16) [reply work/v1/types.go]: fixed

File: cluster/v1/types.go (hunk 2 above)
Comment (qiujian16) [reply cluster/v1/types.go]: fixed

File: cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml (hunk 3 above)
Comment (qiujian16) [reply cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml:38]: fixed

File: cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml (hunk 4 above)
Comment (qiujian16) [reply cluster/v1/0000_00_clusters.open-cluster-management.io_managedclusters.crd.yaml]: fixed

**Comment by qiujian16** [comment]
@pmorie leave to you to review ^^

**Pull request merged**